strawberryfields/cli/__init__.py

```diff
@@ -0,0 +1,219 @@
+
+    # Adding subparsers configure and input
+    subparsers = parser.add_subparsers(title="Commands")
+
+    # Adding the configure subparser
+    configure_parser = subparsers.add_parser(
+        "configure", help="Configure each detail of the API connection."
+    )
+    configure_parser.set_defaults(func=configure)
+
+    configure_parser.add_argument(
+        "--token",
+        "-t",
+        type=str,
+        help="Configure only the token by using defaults for other options.",
+    )
+    configure_parser.add_argument(
+        "--local", "-l", action="store_true", help="Create the configure for the project."
+    )
+
+    # Adding the input subparser
+    run_parser = subparsers.add_parser("run", help="Run a blackbird script.")
+    run_parser.add_argument(
+        "input", type=str, help="The input blackbird script to run.",
+    )
+    run_parser.set_defaults(func=run_blackbird_script)
+    run_parser.add_argument(
+        "--output",
+        "-o",
+        help="Path to the output file, where the results of the program will be written (stdout by default).",
+    )
+
+    return parser
+
+
+def configure(args):
+    """An auxiliary function for configuring the API connection to the Xanadu
+    cloud platform.
+
+    Supports configuration by:
+    * only specifying the token and using default configuration options;
+    * specifying every configuration option one by one.
+
+    Related arguments:
+    * token: the authentication token to use
+    * local: whether or not to create the configuration file locally
+
+    Args:
+        args (ArgumentParser): arguments that were specified on the command
+            line stored as attributes in an argument parser object
+    """
+    if args.token:
+        kwargs = {"authentication_token": args.token}
+    else:
+        kwargs = configure_everything()
+
+    if args.local:
+        store_account(**kwargs, location="local")
+    else:
+        store_account(**kwargs)
+
+
+def ping():
+    """Tests the connection to the remote backend."""
+    if Connection().ping():
+        sys.stdout.write("You have successfully authenticated to the platform!\n")
+    else:
+        sys.stdout.write("There was a problem when authenticating to the platform!\n")
+    sys.exit()
+
+
+def configure_everything():
+    """Provides an interactive selection wizard on the command line to
+    configure every option for the API connection.
+
+    Default configuration options are provided as defaults to the user.
+    Configuration options as detailed in :doc:`/introduction/configuration`.
+
+    Returns:
+        dict[str, Union[str, bool, int]]: the configuration options
+    """
+    default_config = create_config()["api"]
+
+    authentication_token = (
+        input(PROMPTS["authentication_token"].format(default_config["authentication_token"]))
+    )
+
+    if not authentication_token:
+        sys.stdout.write("No authentication token was provided, please configure again.")
+        sys.exit()
```

Comment:
Same question here with `exit()`. Does `return` allow similar behaviour?

Reply:
Here this would be slightly different, as simply having `return` would return to the `main` function and then a call on `store_account` for example would be placed. In the pinging case, it was lying fortunately in a separate branch.

Overall, if the user has chosen to use the configuration wizard, but has not provided an authentication token, then I find it reasonable that the configuration process (and the execution of the script) stops (hence a call on `exit` seems to be reasonable).